File: GdPicture/WinForm/vb.net/Barcode Recognition/Globals.vb
Option Strict On
Option Explicit On

Imports GdPicture14


Enum ThresoldingMode
    Linear
    Bradley
    Otsu
    Sauvola
End Enum

Module Globals
    Public Const DEFAULT_BRADLEY_THRESOLD As Integer = 20 'this is the value internally used in GdPicture to threshold barcode images.
    Public binarizationMode As ThresoldingMode = ThresoldingMode.Bradley

    Public DEFAULT_DESPECKLE_TYPE As DespeckleType = DespeckleType.FourConnected
    Public DEFAULT_DESPECKLE_STRENGTH As Integer = 5

    'Parameters for thresholding functions
    Public thresholdValue As Integer = DEFAULT_BRADLEY_THRESOLD
    Public smoothKernel As Integer 'used for sauvola thresholding
    Public contrastMinDif As Integer 'used for sauvola thresholding

    Public despeckleType As DespeckleType = DEFAULT_DESPECKLE_TYPE
    Public despeckleStrength As Integer = DEFAULT_DESPECKLE_STRENGTH

    Public gdPictureImaging As New GdPictureImaging
    Public gdPicturePDF As New GdPicturePDF
    Public nativeImage As Integer
    Public docSourceType As DocumentType
End Module

File: GdPicture/WinForm/C#/Barcode Recognition/Globals.cs
using GdPicture14;

namespace barcode_recognition
{
    public enum ThresholdingMode
    {
        Linear,
        Bradley,
        Otsu,
        Sauvola,
        Wan
    }

    sealed class Globals
    {
        public const int DEFAULT_BRADLEY_THRESHOLD = 20; //this is the value internally used in GdPicture to threshold barcode images.
        public static ThresholdingMode binarizationMode = ThresholdingMode.Bradley;

        public static DespeckleType DEFAULT_DESPECKLE_TYPE = DespeckleType.FourConnected;
        public static int DEFAULT_DESPECKLE_STRENGTH = 5;

        //Parameters for thresholding functions
        public static int thresholdValue = DEFAULT_BRADLEY_THRESHOLD;
        public static int smoothKernel; //used for sauvola thresholding
        public static int contrastMinDif; //used for sauvola thresholding

        public static DespeckleType despeckleType = DEFAULT_DESPECKLE_TYPE;
        public static int despeckleStrength = DEFAULT_DESPECKLE_STRENGTH;

        public static readonly GdPictureImaging gdPictureImaging = new GdPictureImaging();
        public static readonly GdPicturePDF gdPicturePDF = new GdPicturePDF();
        public static int nativeImage;
        public static DocumentType docSourceType;
    }
}
